src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1398,6 +1435,75 @@ private MqttServerStream resolvePublisher(
             return stream;
         }
 
+        private void decodeSession(
+            long traceId,
+            long authorization,
+            int flags)
+        {
+            final String topic = String.format(SESSION_WILDCARD_TOPIC_FORMAT, clientId);
+            final int topicKey = topicKey(topic);
+
+            final RouteFW route = resolveTarget(routeId, authorization, topic,  PUBLISH_ONLY);
+
+            MqttServerStream stream;
+            if (route != null)
+            {
+                final long resolvedId = route.correlationId();
+
+                final JsonObject sessionObj = OBJECT_BUILDER.add(SESSION_EXPIRES_AT_NAME, sessionExpiresAt)
+                                                            .add(WILL_DELAY_NAME, mqttConnectPayloadRO.willDelay)
+                                                            .build();
+
+                final OctetsFW payload = sessionPayloadRW.wrap(payloadBuffer, 0, payloadBuffer.capacity())
+                                                         .set(sessionObj.toString().getBytes(UTF_8))
+                                                         .build();
+
+                final MqttDataExFW dataEx = mqttDataExRW.wrap(dataExtBuffer, 0, dataExtBuffer.capacity())
+                                                        .typeId(mqttTypeId)
+                                                        .topic(String.format(SESSION_TOPIC_FORMAT, clientId))
+                                                        .build();
+
+                stream = streams.computeIfAbsent(topicKey, s -> new MqttServerStream(resolvedId, 0, topic));
+
+                final int reserved = payload.sizeof() + mqttConnectPayloadRO.willPayload.sizeof() + stream.initialPadding;
+
+                stream.doApplicationBeginOrFlush(traceId, authorization, affinity, topic, NO_FLAGS, 0, PUBLISH_AND_SUBSCRIBE);
+                stream.doApplicationData(traceId, authorization, reserved, payload, dataEx);
```

Comment:
This needs to be done in two steps, one to extract the `clientId` and initiate the application stream, then another to process the resulting `WINDOW` to trigger sending of the application data, after first acquiring the shared budget for both client and will messages as needed.

If the application stream receives a `RESET`, then we need to handle it the same way as if there is no route, with something like an `AmqpSessionStateStream` that will echo back whatever is published.